src/pages/UserRegister.vue

```diff
@@ -1,15 +1,407 @@
+        return true;
+      return false;
+    },
+    firstNameError() {
+      if (!this.firstName && !this.firstNameTouched) return "";
+      if (!this.firstName && this.firstNameTouched)
+        return "Please enter your first name";
+      if (!regex.name.test(this.firstName))
+        return "Please enter correct first name";
+      return "";
+    },
+    lastNameError() {
+      if (!this.lastName && !this.lastNameTouched) return "";
+      if (!this.lastName && this.lastNameTouched)
+        return "Please enter your last name";
+      if (!regex.name.test(this.lastName))
+        return "Please enter correct last name";
+      return "";
+    },
+    birthDayError() {
+      if (!this.birthDay && !this.birthDayTouched) return "";
+      if (!this.birthDay && this.birthDayTouched) return "error";
+      if (!isAdult(this.birthDay)) return "error";
+      return "";
+    },
+    isFormValid() {
+      if (
+        this.email &&
+        !this.emailError &&
+        this.password &&
+        !this.passwordError &&
+        this.firstName &&
+        !this.firstNameError &&
+        this.lastName &&
+        !this.lastNameError &&
+        this.birthDay &&
+        !this.birthDayError &&
+        this.privacyPolicy
+      )
+        return true;
+      return false;
+    },
+  },
+  methods: {
+    togglePasswordInputType() {
+      return this.passwordInputType === "password"
+        ? (this.passwordInputType = "text")
+        : (this.passwordInputType = "password");
+    },
+    getValidationClass(state, stateTouched, error) {
+      if (!state && !stateTouched) return "";
+      if (!state && stateTouched) return "error";
+      if (state && error) return "error";
+      if (state && !error) return "correct";
+    },
+    resetState() {
+      this.email = "";
+      this.emailTouched = false;
+      this.password = "";
+      this.passwordTouched = false;
+      this.firstName = "";
+      this.firstNameTouched = false;
+      this.lastName = "";
+      this.lastNameTouched = false;
+      this.birthDay = "";
+      this.birthDayTouched = false;
+      this.privacyPolicy = false;
+    },
+    onSubmit() {
+      if (this.isFormValid) {
+        // send data to api
+        this.resetState();
+        this.$router.back();
+      } else {
+        return;
+      }
+    },
+  },
 });
 </script>
 
 <template>
-  <div>
-    <h1>Register form page</h1>
-  </div>
+  <section class="register">
+    <SectionContainer class="register__container">
+      <SectionTitleSecondary
+        v-if="step === 1"
+        title="Ahoy you!"
+        subtitle="Care to register?"
+        class="register__title"
+      />
+      <SectionTitleSecondary
+        v-if="step === 2"
+        title="Great!"
+        subtitle="Now your name"
+        class="register__title"
+      />
+
+      <form class="register__form" novalidate @submit.prevent="onSubmit">
+        <div class="register__step">
+          <div v-if="step === 1" class="register__step-inputs">
```

Comment:
There is some potential here to slim down this large component by extracting each step template and logic to it's own component :)

Reply:
Initially I wanted to create two separate components with the first and second step, but I think I don’t have enough buttons :).

So I have to send all the data together to the register endpoint to create a new user, so I have to have one form.
Let’s assume I nest two components into a form tag: 
- RegisterFirstStep - with email, password, next button - the next button will be used to emit data to the parent component and increment step.
- RegisterSecondStep - with firstname, lastname, date of birth, privacy policy and register button - if I use the register button as a button to emit data to the parent component, the parent component has all the data, but I don’t have a submit button anymore. 
As far as I know if a button is of submit type, I cannot have onclick event attached to this button. If I want some additional action to be done by this submit button I can put it into submit function, but the submit function will be in the parent component.
So I just don’t know how to handle registration steps components with passing data to the parent having the buttons provided in the design.

Reply:
Your overview is accurate. Since you basically replace form content and handle all the state in js how about creating two forms instead, one for each step. That way it's easier to separate those two and you get advantage of handling submit instead button click in both of them. You still can provide submit handler from the parent component but your logic related to a specific step like validation can be moved accordingly

Reply:
I was not familiar with this way of handling forms and its submit, but asked Szymon for some explanation and then I managed to create two separate steps for register form. So let me know if you accept this solution.

Reply:
Looks great, nicely done